### DISCUSSION TITLE
collection/hashmap

### DISCUSSION THREAD

**Discussion by giscus** [itwanger/tobebetterjavaer-giscus#104] [Announcements]
# collection/hashmap

Java程序员进阶之路，小白的零基础Java教程，Java HashMap详解

https://tobebetterjavaer.com/collection/hashmap.html

**Comment by wylgithub** [comment]
一枚用心的IT大哥 很强！

**Reply by itwanger** [reply]
希望能有所帮助

**Comment by 2258324319** [comment]
太底了，日后再说

**Reply by itwanger** [reply]
这是什么虎狼之词（😂）

**Comment by shallowgit** [comment]
例如，-10 mod 3 的结果是 -1，而 -10 % 3 的结果是 2，因为 -10 除以 3 的余数是 -1，所以 -10 取模 3 的结果应该是 -1；而 -10 对 3 取余的结果是 2，因为：-10 ÷ 3 = -3 余 -1，由于除数为正数 3，余数的符号应与被除数的符号相同，因此余数应为正数 2，而不是 -1。

这怎么突然蹦出来个2？打错了吧

**Reply by shallowgit** [reply]
搜了下取余的计算，我都看蒙了

**Comment by jackttj** [comment]
大佬太强了 大佬总结出得这些事在工作经验中总结出来的 还是自己在学习的过程中总结的

**Comment by Eivae** [comment]
取模、取余运算那部分写错了吧，文字部分的结果和代码部分的结果刚好是反的，代码那部分才是对的吧

**Reply by Eivae** [reply]
还有，%不是取余吗，怎么变成取模了？取模不是用math.floorMod()吗？

**Comment by Eivae** [comment]
讲加载因子那一节，算概率那个公式，根本就是错的吧？他用的是n重伯努利实验或者二项分布的公式，但这样的前提是一次实验只有两个结果（就像投一次硬币只有正或反两种结果一样），而且两次实验之间的结果要毫无关系。但这里的问题是前一个结果显然会对后面的结果产生影响，怎么能代这个公式呢。

**Comment by Ricehahaa** [comment]
好蓝，看不懂

**Comment by Leslie-DD** [comment]
太顶了

**Comment by BOUNDLESS-558** [comment]
有个地方的数值算错了吧。

将 <code>00000000 10100101 11000100 00100101</code> 右移 16 位得到的结果是 <code>00000000 00000000 00000000 10100101</code>，然后和 <code>00000000 10100101 11000100 00100101 </code>进行异或操作，步骤如下。

00000000 10100101 11000100 00100101
00000000 00000000 00000000 10100101
00000000 10100101 11000100 10000000 

再将 00000000 10100101 11000100 10000000  和 00000000 00000000 00000000 00001111 做取模运算（&）运算步骤如下。

00000000 10100101 11000100 10000000 
00000000 00000000 00000000 00001111
00000000 00000000 00000000 00000000 为 0。

**Reply by BOUNDLESS-558** [reply]
但是，作者原文当中提到的将  00000000 00000000 00000000 10100101 和 00000000 10100101 11000100 00100101  进行异或得到的值是 00000000 10100101 00111011 10000000？

**Reply by withdong02** [reply]
我觉得你是对的
